src/Libraries/CoreNodes/List.cs

```diff
@@ -688,6 +689,78 @@ public static IList Transpose(IList lists)
             return transposedList;
         }
 
+        /// <summary>
+        /// Checks if a given list is empty or is a single dimensional list that contains only null values
+        /// </summary>
+        /// <param name="list"></param>
+        /// <returns>Returns true if the input list is empty or contains only null values</returns>
+        public static bool IsNullOrEmpty(IList list)
+        {
+            return list.Cast<object>().All(el => el == null);
+        }
+
+        /// <summary>
+        /// Cleans data of nulls and empty lists from a given list of arbitrary dimension
+        /// </summary>
+        /// <param name="list"></param>
+        /// <param name="preserveIndices">Provide an option to preserve the indices of the data
+        /// such that non-trailing nulls may not be filtered out</param>
+        /// <returns>A list cleaned of nulls and empty lists</returns>
+        public static IList Clean(IList list, bool preserveIndices = true)
+        {
+            if (list.Count == 0)
+                return list;
+
+            var culledList = new List<object>();
+            if (preserveIndices)
+            {
+                if (IsNullOrEmpty(list))
+                    return null;
```

Comment:
expected? I mean, `{1, null}` returns `{1}`, `{null}` returns `{}`, and `{}` returns null? (BTW, unit test needs to cover this edge conditions)....